datafusion/common/src/scalar/mod.rs

```diff
@@ -99,6 +102,66 @@ use arrow_buffer::NullBuffer;
 /// # }
 /// ```
 ///
+/// # Nested Types
+///
+/// `List` / `LargeList` / `FixedSizeList` / `Struct` are represented as a
+/// single element array of the corresponding type.
+///
+/// ## Example: Creating [`ScalarValue::Struct`] using [`ScalarStructBuilder`]
+/// ```
+/// # use std::sync::Arc;
+/// # use arrow::datatypes::{DataType, Field};
+/// # use datafusion_common::{ScalarValue, scalar::ScalarStructBuilder};
+/// // Build a struct like: {a: 1, b: "foo"}
+/// let field_a = Field::new("a", DataType::Int32, false);
```

Comment:
The use case for the `ScalarStructBuilder` is pretty nicely illustrated by this example compared to the one in

```
## Example: Creating [`ScalarValue::Struct`] directly
```